ui_framework/generator-kui/documentation/index.js

```diff
@@ -0,0 +1,216 @@
+
+      const path = DOCUMENTATION_PAGE_PATH;
+
+      const vars = config.documentationVars = {
+        componentExampleName,
+        fileName,
+      };
+
+      const sandboxPath
+        = config.documentationPageDemoPath
+        = `${path}/${config.name}/${fileName}`;
+
+      this.fs.copyTpl(
+        this.templatePath('documentation_sandbox.html'),
+        this.destinationPath(`${sandboxPath}_sandbox.html`)
+      );
+
+      this.fs.copyTpl(
+        this.templatePath('documentation_sandbox.js'),
+        this.destinationPath(`${sandboxPath}_sandbox.js`),
+        vars
+      );
+    };
+
+    switch (this.fileType) {
+      case 'documentation':
+        writeDocumentationPage();
+        writeDocumentationPageDemo(config.name, config.name);
+        break;
+
+      case 'demo':
+        writeDocumentationPageDemo(config.demoName, config.folderName);
+        break;
+
+      case 'sandbox':
+        writeSandbox();
+        break;
+    }
+  }
+
+  end() {
+    const showImportDemoSnippet = () => {
+      const {
+        componentExampleName,
+        componentExamplePrefix,
+        fileName,
+      } = this.config.documentationVars;
+
+      this.log(chalk.white('\n// Import demo into example.'));
+      this.log(
+        `${chalk.magenta('import')} ${componentExampleName} from ${chalk.cyan(`'./${fileName}'`)};\n` +
+        `${chalk.magenta('const')} ${componentExamplePrefix}Source = require(${chalk.cyan(`'!!raw!./${fileName}'`)});\n` +
+        `${chalk.magenta('const')} ${componentExamplePrefix}Html = renderToHtml(${componentExampleName});`
+      );
+
+      this.log(chalk.white('\n// Render demo.'));
+      this.log(
+        `<GuideSection\n` +
+        `  title="${componentExampleName}"\n` +
+        `  source={[{\n` +
+        `    type: GuideSectionTypes.JS,\n` +
+        `    code: ${componentExamplePrefix}Source,\n` +
+        `  }, {\n` +
+        `    type: GuideSectionTypes.HTML,\n` +
+        `    code: ${componentExamplePrefix}Html,\n` +
+        `  }]}\n` +
+        `>\n` +
+        `  <GuideText>\n` +
+        `    Description needed: how to use the ${componentExampleName} component.\n` +
+        `  </GuideText>\n` +
+        `\n` +
+        `  <GuideDemo>\n` +
+        `    <${componentExampleName} />\n` +
+        `  </GuideDemo>\n` +
+        `</GuideSection>\n`
+      );
+    };
+
+    const showImportRouteSnippet = suffix => {
+      const {
+        componentExampleName,
+        fileName,
+      } = this.config.documentationVars;
+
+      this.log(chalk.white('\n// Import example into routes.js.'));
+      this.log(
+        `${chalk.magenta('import')} ${componentExampleName}${suffix}\n` +
+        `  ${chalk.magenta('from')} ${chalk.cyan(`'../../views/${fileName}/${fileName}_${suffix.toLowerCase()}'`)};`
+      );
+
+      this.log(chalk.white('\n// Import route definition into routes.js.'));
+      this.log(
+        `{\n` +
+        `  name: ${chalk.cyan(`'${componentExampleName}'`)},\n` +
```

Comment:
The proposed route name collides with that of the component docs. If the user just copy+pastes this the sandbox menu entry will link to the component doc page. Maybe we could propose something with a prefix or suffix to avoid that?